src/logic/hooks/useButtonStatus.tsx

```diff
@@ -0,0 +1,29 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react'
+
+import { EstimationStatus } from './useEstimateTransactionGas'
+import { ButtonStatus } from 'src/components/Modal'
+
+export const useButtonStatus = (
+  data?: string,
+  txEstimationStatus?: EstimationStatus,
```

Comment:
This implementation seems to be tightly coupled with the EstimationStatus, so maybe it should be reflected in the hook name? Like `useEstimationStatus`?

A more generic hook would be probably just taking a promise.
If it's null, the button is in the default state, if it's not resolved/rejected yet – it's loading.

Reply:
ok, changed the name to `useEstimationStatus` and also removed the `data` parameter (it wasn't used, nor necessary).

It feels a bit excessive the use of a hook now, but I kept it as it's easy to update the status with the `setButtonStatus` action.

---

After getting rid of the `data` param, I realize that we were lacking validation in the data field for the Custom Contract Interaction, so I fixed that too ([`2e44eda` (#2247)](https://github.com/gnosis/safe-react/pull/2247/commits/2e44edaddd47ecc52c5baf5889da9b2595c49ba9))